src/js/features/conversations.js

```diff
@@ -34,6 +33,9 @@ function Conversations(timeout) {
     if (window.App && App.__container__.lookup('service:whispers-shim')) {
         var whisperShim = App.__container__.lookup('service:whispers-shim');
         whisperShim.on('whisper', _self.onWhisper);
+        whisperShim.on('thread', function(data) {
+            bttv.ws.joinConversation(data.id);
```

Comment:
the "whipser" event is only called when you **receive** one, I couldn't find a way to hook on whisper sent. That being sent, this "thread" event is called every time you **read**, so technically, it's equivalent, and better so if someone spams you 20 whispers, you don't broadcast 20 times, only once when you click on the conversation.

As for parting, I'll try to find something for the window closing but it might be hack-ish.